server/util/clickhouse/schema/schema.go

```diff
@@ -196,8 +206,12 @@ type Execution struct {
 	StatusCode int32
 	ExitCode   int32
 
-	CachedResult bool
-	DoNotCache   bool
+	CachedResult    bool
+	DoNotCache      bool
+	SkipCacheLookup bool
+
+	ExecutionPriority int32
+	IsolationType     string
```

Comment:
I think this would be a great use case for ClickHouse's `LowCardinality(String)` column type - it builds a dictionary to represent the column values as an int, while still allowing the column to be queried as a human readable string. Though idk how to annotate this field so that gorm applies the `LowCardinality` option. (Also, we'd want to check for a known set of values when storing the field, to enforce the low cardinality.)

Reply:
I was considering that, but I also couldn't figure out how to use it with gorm. I'll take another look.

Reply:
Turns out that it's pretty simple. I couldn't find any clear documentation, but gorm supports specifying the type as a [field tag](https://gorm.io/docs/models.html#Fields-Tags), and as long as you specify a clickhouse compatible type, the clickhouse gorm driver respects this. I verified it by running locally and querying the system schema for the field type.

Reply:
As far as enforcing low cardinality, I don't think we should fail execution when we see a new isolation type, so I think we're left with a few options:

1. Keep track of the observed isolation types in the app, and log or report a metric of the count.
2. Monitor an existing metric, like [buildbuddy_remote_execution_count](https://github.com/buildbuddy-io/buildbuddy/blob/1ff042e870fdfae3b32eefc148d2b8bb98f278b0/server/metrics/metrics.go#L823-L831), which has an isolation type label, and alert on too many types.
3. Query the clickhouse database periodically and somehow report if the cardinality is too high. I'm not sure of the best way to do this, but I think clickhouse might have a built in way to turn queries into metrics.

Regardless, I don't think we have to have this enforcement right away.

Reply:
another option is to have a hard-coded list of isolation types (`platform.go` is probably the most convenient place for it) and then coerce any values not in that list to an empty string or the string "unknown"

Reply:
We chatted and agreed to not use LowCardinality for the user requested isolation type. It should usually be empty, and it might be useful to analyze values which don't belong to a known set.

For the effective isolation type, we can mostly trust the executor, but we decided its better to force the value to belong to a known set, so clickhouse query performance doesn't become worse if we make a mistake.